cfme/storage/manager.py

```diff
@@ -0,0 +1,233 @@
+# -*- coding: utf-8 -*-
+
+from navmazing import NavigateToAttribute
+from widgetastic.utils import Version, VersionPick
+from widgetastic.widget import View, NoSuchElementException, Text
+from widgetastic_manageiq import (
+    Accordion,
+    BootstrapSelect,
+    BootstrapTreeview,
+    BreadCrumb,
+    ManageIQTree,
+    PaginationPane,
+    SummaryTable,
+    Table
+)
+from widgetastic_patternfly import (
+    Button,
+    Dropdown,
+    FlashMessages
+)
+
+from cfme.base.ui import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from cfme.utils.appliance import BaseCollection, BaseEntity
+
+
+class ManagerToolbar(View):
+    """The toolbar on the Storage Manager or Provider page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+
+
+class ManagerDetailsToolbar(View):
+    """The toolbar on the Storage Manager or Provider detail page"""
+    reload = Button(title='Reload Current Display')
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    monitoring = Dropdown('Monitoring')
+    download = Button(title='Download summary in PDF format')
+
+
+class ManagerEntities(View):
+    """The entities on the main list Storage Manager or Provider page"""
+    table = Table(".//div[@id='list_grid']/table")
+
+
+class ManagerDetailsEntities(View):
+    """The entities on the Storage Manager or Provider details page"""
+    breadcrumb = BreadCrumb()
+    properties = SummaryTable('Properties')
+    relationships = SummaryTable('Relationships')
+    smart_management = SummaryTable('Smart Management')
+    status = SummaryTable('Status')
+
+
+class ManagerDetailsAccordion(View):
+    """The accordion on the Storage Manager or Provider details page"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class ManagerView(BaseLoggedInPage):
+    """A base view for all the Storage Manager or Provider pages"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]')
+
+    @property
+    def in_manager(self):
+        nav = self.context['object'].nav.pick(self.context['object'].appliance.version)
```

Comment:
REQUIRED: this is good idea to move navigation path to class but  I don't like this name - "nav". It doesn't explain what it contains even more it sounds similar to navmazing object/property. I'd advice to call it nav_path or navigation_path for example.